script.js

```diff
@@ -11,29 +71,47 @@ function createListItem(taskContent) {
 
   return listItem;
 }
-
+//////////////////////////////
 if (listContainer) {
-  listContainer.addEventListener("click", handleListClick, false);
+  listContainer.addEventListener("click", handleListClick);
 }
 
 function handleListClick(event) {
   const clickedElement = event.target;
 
   if (clickedElement.tagName === "LI") {
     clickedElement.classList.toggle("checked");
+    tasks.map((task) => {
+      if (task.id === +clickedElement.id) {
+        task.status = task.status === "active" ? "completed" : "active";
+      }
+      return task;
+    });
   } else if (clickedElement.tagName === "SPAN") {
     clickedElement.parentElement.remove();
   }
 }
```

Comment:
Ensure that the logic for checking status and removing tasks is simplified.

// Modified Function: Simplifying event logic
function handleListClick(event) {
  const clickedElement = event.target;

  if (clickedElement.tagName === "LI") {
    clickedElement.classList.toggle("checked");
    tasks.map((task) => {
      if (task.id === +clickedElement.id) {
        task.status = task.status === "active" ? "completed" : "active";
      }
      return task;
    });
  } else if (clickedElement.tagName === "SPAN") {
    // New Logic: Ensuring task removal from array
    tasks = tasks.filter(task => task.id !== +clickedElement.parentElement.id);
    clickedElement.parentElement.remove();
  }
}